services/community-site/src/pages/Delegation/NodePage.tsx

```diff
@@ -0,0 +1,313 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { toSvg } from 'jdenticon';
+
+import { Button, Checkbox, Icons, Modal } from '@taraxa_project/taraxa-ui';
+import { useMetaMask } from 'metamask-react';
+
+import { useAuth } from '../../services/useAuth';
+
+import Title from '../../components/Title/Title';
+import { useDelegationApi } from '../../services/useApi';
+import Delegation from '../../interfaces/Delegation';
+import PublicNode from '../../interfaces/PublicNode';
+
+import './nodePage.scss';
+import Delegate from './Modal/Delegate';
+import Undelegate from './Modal/Undelegate';
+import CloseIcon from '../../assets/icons/close';
+
+const NodePage = () => {
+  const auth = useAuth();
+  const { status, account } = useMetaMask();
+  const [availableBalance, setAvailableBalance] = useState(0);
+  const [delegationAtTop, setDelegationAtTop] = useState<boolean>(false);
+  const [node, setNode] = useState<PublicNode | null>(null);
+  const [delegationCount, setDelegationCount] = useState<number>(0);
+  const [delegations, setDelegations] = useState<Delegation[] | []>([]);
+  const [delegationPage, setDelegationPage] = useState<number>(1);
+  const [delegateToNode, setDelegateToNode] = useState<PublicNode | null>(null);
+  const [undelegateFromNode, setUndelegateFromNode] = useState<PublicNode | null>(null);
+  const { nodeId } = useParams<{ nodeId?: string }>();
+  const delegationApi = useDelegationApi();
+  const nodeIcon = toSvg('Aweesome node 1', 40, { backColor: '#fff' });
+
+  const isLoggedIn = !!auth.user?.id;
+  const canDelegate = isLoggedIn && status === 'connected' && !!account;
+
+  const getBalances = useCallback(async () => {
+    if (!canDelegate) {
+      return;
+    }
+    const data = await delegationApi.get(`/delegations/${account}/balances`);
+    if (data.success) {
+      setAvailableBalance(data.response.remaining);
+    }
+  }, [canDelegate, account]);
+
+  const fetchNode = useCallback(async () => {
+    const data = await delegationApi.get(`/validators/${nodeId}`);
+    if (data.success) {
+      setNode(data.response);
+      // eslint-disable-next-line no-console
+      console.log(data.response);
+      // eslint-disable-next-line no-console
+      console.log(node);
+    }
+  }, [nodeId]);
+
+  const fetchDelegators = useCallback(async () => {
+    setDelegations([]);
+    const data = await delegationApi.get(
+      `/validators/${nodeId}/delegations?show_my_delegation_at_the_top=${delegationAtTop}&page=${delegationPage}`,
+    );
+    if (data.success) {
+      setDelegations(data.response.data);
+      setDelegationCount(data.response.count);
+    }
+  }, [nodeId, delegationAtTop, delegationPage]);
+
+  useEffect(() => {
+    getBalances();
+    fetchNode();
+    fetchDelegators();
+  }, [fetchNode, fetchDelegators]);
```

Comment:
Maybe we should also add `getBalances` to the dep list so we don't get a warning.